### PR TITLE
feat: Add api to delete sboms

### PR DESCRIPTION
**Pull request by chirino**
A task of #563

### CODE REVIEW

**Review by ctron — DISMISSED** [review]
Looks good to me. Except for the commented out code :) … we should try to have less of that with every PR.

**Comment by chirino** [comment]
@ctron I think the parts that are commented out should eventually get uncommented, it's commented right now because uncommented the unit tests fail.   Do we have any examples of API tests that use authenticated API endpoints?

**Comment by chirino** [comment]
Not sure what that means to be honest.  Could that just be done in a subsequent Pr to enable and test authentication?

**Comment by jcrossley3** [comment]
> Could that just be done in a subsequent Pr to enable and test authentication?

No. When you pull in the latest changes to resolve the conflicts on this branch, you'll see that `with_upload` is gone. And you'll see examples of what the other tests are doing without it.

I didn't realize there were pending uses of `with_upload` when I merged it. Sorry about that.

File: modules/fundamental/src/sbom/endpoints/test.rs [hunk 1]
```diff
@@ -149,3 +150,42 @@ async fn set_labels_not_found(ctx: &TrustifyContext) -> Result<(), anyhow::Error
     })
     .await
 }
+
+/// Test deleting an sbom
+#[test_context(TrustifyContext)]
+#[test(actix_web::test)]
+async fn delete_sbom(ctx: &TrustifyContext) -> Result<(), anyhow::Error> {
+    with_upload(ctx, |result, app| {
+        Box::pin(async move {
```
Comment (jcrossley3) [inline modules/fundamental/src/sbom/endpoints/test.rs:159]:
```suggestion
    let app = caller(ctx).await?;
    let result = ctx
        .ingest_document("quarkus-bom-2.13.8.Final-redhat-00004.json")
        .await?;
```

File: modules/fundamental/src/sbom/endpoints/test.rs [hunk 2]
```diff
@@ -149,3 +150,42 @@ async fn set_labels_not_found(ctx: &TrustifyContext) -> Result<(), anyhow::Error
...
+                )
+                .await;
+
+            log::debug!("Code: {}", response.status());
+            assert_eq!(response.status(), StatusCode::NOT_FOUND);
+
+            Ok(())
+        })
+    })
+    .await
```
Comment (jcrossley3) [inline modules/fundamental/src/sbom/endpoints/test.rs:190]:
```suggestion
```

File: modules/fundamental/src/sbom/endpoints/mod.rs [hunk 3]
```diff
@@ -190,6 +192,39 @@ pub async fn get(
...
+    responses(
+        (status = 200, description = "Matching SBOM", body = SbomDetails),
+        (status = 404, description = "Matching SBOM not found"),
+    ),
+)]
+#[delete("/v1/sbom/{id}")]
+pub async fn delete(
+    service: web::Data<SbomService>,
+    // authorizer: web::Data<Authorizer>,
+    // user: UserInformation,
```
Comment (jcrossley3) [inline modules/fundamental/src/sbom/endpoints/mod.rs:210]:
```suggestion
    authorizer: web::Data<Authorizer>,
    user: UserInformation,
```

File: modules/fundamental/src/sbom/endpoints/mod.rs [hunk 4]
```diff
@@ -190,6 +192,39 @@ pub async fn get(
...
+    id: web::Path<Id>,
+) -> actix_web::Result<impl Responder> {
+    // authorizer.require(&user, Permission::DeleteSbom)?;
```
Comment (jcrossley3) [inline modules/fundamental/src/sbom/endpoints/mod.rs:213]:
```suggestion
    authorizer.require(&user, Permission::DeleteSbom)?;
```

**Review by jcrossley3 — CHANGES_REQUESTED** [review]
Once you merge the latest changes...

**Comment by chirino** [comment]
@jcrossley3 thanks!  rebased.

**Review by jcrossley3 — APPROVED** [review]

**Pull request merged**